python-sdk/src/astro/custom_backend/serializer.py

```diff
@@ -43,7 +43,7 @@ def _attempt_to_serialize_unknown_object(obj: object):
     try:
         return json.dumps(obj)
     except TypeError:
-        return pickle.dumps(obj).hex()
+        return obj
```

Comment:
Why are we returning the `obj` when `TypeError`?

Reply:
Because if we can't serialize the object, we have to leave it to the user to decide whether they want to turn on pickling or not. The entire point of this backend is to remove the pickling dependency and I think people wouldn't be stoked if we were pickling objects and not telling them.

Reply:
in that case, should we just do `json.dumps(obj)` in `_attempt_to_serialize_unknown_object`? so that it will blow up for unknown type and they know that it is because of the serialization error?

Reply:
where are we uploading to a given storage url here?